### PR TITLE
Bounty challenge claiming

### PR DESCRIPTION
**Pull request by Nicoslz**
This PR introduces the feature of claiming a bounty's challenge. In the challenge display section for a specific challenge, users now have the option to click on the "Claim" button, enabling them to stake a claim for the selected challenge.

Upon clicking the "Claim" button, users will be redirected to their camera interface to capture a photo. Subsequently, they will be navigated to a new view where they can finalize their claim.

This PR resolves issue #199 and is dependent on related issues #211, #212, and #213.

### CODE REVIEW

File: app/src/main/java/com/github/geohunt/app/ui/components/navigation/NavigationController.kt [hunk 1]
```diff
@@ -199,5 +200,31 @@ fun NavigationController(
...
+                bid = bid,
+                cid = cid,
+                onFailure = {
+                    Toast.makeText(context, "Something went wrong, failed to create challenge", Toast.LENGTH_LONG).show()
+                    Log.e("GeoHunt", "Fail to create challenge: $it")
+                    navController.popBackStack()
+                },
+                onClaimSubmitted = {
+                    navController.popBackStack()
+                    navController.navigate("challenge-view/$cid")
```
Comment (Maeeen) [inline app/src/main/java/com/github/geohunt/app/ui/components/navigation/NavigationController.kt:225]: Use the hidden route?

**Review by Maeeen — APPROVED** [review]
Overall good code except a few caveats that are really minor.

**Review by Maeeen — DISMISSED** [review]
Overall good code except a few caveats that are really minor.

File: app/src/main/java/com/github/geohunt/app/ui/components/bounties/BountyClaimChallenge.kt [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+                verticalArrangement = Arrangement.Center,
+                horizontalAlignment = Alignment.CenterHorizontally,
+                modifier = Modifier.verticalScroll(rememberScrollState())
+            ) {
+                Image(
+                    painter = bitmapPainter,
+                    modifier = Modifier
+                        .aspectRatio(bitmapPainter.intrinsicSize.width / bitmapPainter.intrinsicSize.height)
+                        .fillMaxSize(0.5f),
+                    contentDescription = "Photo just taken of the claim"
```
[the remark this answers is not in the document]
Comment (Nicoslz) [reply app/src/main/java/com/github/geohunt/app/ui/components/bounties/BountyClaimChallenge.kt:66]: I have placed this string in strings.xml

File: app/src/main/java/com/github/geohunt/app/ui/components/bounties/BountyClaimViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+import java.io.File
+
+class BountyClaimViewModel(
+    private val bountyId: String,
+    private val imageRepository: ImageRepository,
+    private val locationRepository: LocationRepository,
+    private val challengeRepository: ChallengeRepositoryInterface,
+    private val bountyClaimRepositoryInterface: BountyClaimRepositoryInterface,
```
[the remark this answers is not in the document]
Comment (Nicoslz) [reply app/src/main/java/com/github/geohunt/app/ui/components/bounties/BountyClaimViewModel.kt:31]: You are right, placing the Interface at the end of the name is a bit redundant. I have changed the name.

File: app/src/androidTest/java/com/github/geohunt/app/ui/components/bounties/BountyClaimChallengeTest.kt [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+        val bitmap = createTestBitmap(context)
+        val awaitingAnimation = CompletableDeferred<Unit>()
+
+        composeTestRule.setContent {
+            BountyChallengeSubmitClaimForm(bitmap = bitmap, state = BountyClaimViewModel.State.READY_TO_CLAIM) {
+                awaitingAnimation.complete(Unit)
+            }
+        }
+
+        composeTestRule.onNodeWithText("Submit Claim")
```
[the remark this answers is not in the document]
Comment (Nicoslz) [reply app/src/androidTest/java/com/github/geohunt/app/ui/components/bounties/BountyClaimChallengeTest.kt:76]: In my opinion the node looks better with "Submit Claim" than with "Submit claim". At the end it's absolutely minor

File: app/src/main/java/com/github/geohunt/app/ui/components/navigation/NavigationController.kt [hunk 5]
```diff
@@ -199,5 +200,31 @@ fun NavigationController(
             )
             PrivacySettingsPage(onBack = { navController.popBackStack() }, viewModel)
         }
+
+        // Bounties
+        // Open a claim for a given bounty's challenge
+        composable(
+            "bounty-claim-challenge/{bountyId}/{challengeId}",
```
[the remark this answers is not in the document]
Comment (Nicoslz) [reply app/src/main/java/com/github/geohunt/app/ui/components/navigation/NavigationController.kt:228]: This is a very good claim. I did not think about this before. I have changed my code now to use the ENUM.

File: app/src/main/java/com/github/geohunt/app/ui/components/navigation/NavigationController.kt (hunk 1 above)
Comment (Nicoslz) [reply app/src/main/java/com/github/geohunt/app/ui/components/navigation/NavigationController.kt:246]: As above, I have changed my code to use the hidden route.

**Pull request merged**